.github/actions/sls-package/action.yml

```diff
@@ -0,0 +1,79 @@
+name: sls package action
+description: Reusable sls package
+
+inputs:
+  environment-name:
+    description: 'The environment name'
+    required: true
+  app:
+    description: 'The application - crn-server or gp2-server'
+    required: true
+  aws-access-key-id:
+    description: 'AWS Access Key ID'
+    required: true
+  aws-secret-access-key:
+    description: 'AWS Secret Access Key'
+    required: true
+  lightstep-token:
+    description: 'Lightstep Token'
+    required: true
+  squidex-client-secret:
+    description: 'Squidex Client Secret'
+    required: true
+  squidex-shared-secret:
+    description: 'Squidex Shared Secret'
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Setup
+      id: setup
+      uses: ./.github/actions/setup-environment
+      with:
+        environment-name: ${{ inputs.environment-name }}
+    - name: Cache build output
+      uses: ./.github/actions/cache-unplugged
+      with:
+        environment-name: ${{ inputs.environment-name }}
+    - name: Cache build definitions
+      uses: ./.github/actions/cache-build
+      with:
+        cache-prefix: ${{ steps.setup.outputs.cache-prefix-definitions }}
+        environment-name: ${{ inputs.environment-name }}
+    - name: Cache transpile output
+      uses: ./.github/actions/cache-build
+      with:
+        cache-prefix: ${{ steps.setup.outputs.cache-prefix-transpile }}
+        environment-name: ${{ inputs.environment-name }}
+    - name: Generate package
+      shell: bash
+      run: |
+        yarn workspace @asap-hub/${{ inputs.app }} run pnpify serverless package --package $PWD/${{ inputs.app }}-sls-package
+      env:
+        ALGOLIA_INDEX: ${{ steps.setup.outputs.algolia-index }}
```

Comment:
is it not a problem that we are having to add all the environment variables for gp2 and crn regardless of which one we are packaging? there is no algolia or squidex in gp2 and yet its all here in the actions

Reply:
@markcsims